src/main/java/pages/LoginPageFactory.java

```diff
@@ -0,0 +1,54 @@
+package pages;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class LoginPageFactory extends BasePage {
+    public LoginPageFactory(WebDriver driver) {
+        super(driver);
+    }
+
+    @Override
+    BasePage waitForPageOpened() {
+        return null;
+    }
+
+    @FindBy(xpath="//*[@data-test='username']")
+    WebElement usernameInput;
+    @FindBy(xpath="//*[@data-test='password']")
+    WebElement userpasswordInput;
+
+    @FindBy(id = "login-button")
+    WebElement loginBtn;
```

Comment:
better to move findBy's above all methods in class. and delete extra space between userpasswordInput and loginBtn

Reply:
done